app/posts/releasing-rails-libraries-for-version-5.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Releasing the Rails libraries for version 5 of the GOV.UK Design System
+description: Notes on how the Rails libraries for the GOV.UK Design System are developed and the work required to support version 5
+date: 2024-01-10
+author:
+  name: Peter Yates
+  url: https://github.com/peteryates
+---
+
+Version 5 of the GOV.UK Design System was released on December 8th 2023. Half an hour later, we released version 5 of the x-govuk [form builder](https://govuk-form-builder.netlify.app/) and [components](https://govuk-components.netlify.app/) libraries.
+
+The short turnaround time might suggest it was a quick job, but work on the included features started in May and we've made some significant improvements since then.
+
+For the uninitiated, the form builder and components libraries are implementations of the [GOV.UK Design System](https://design-system.service.gov.uk/) for [Ruby on Rails](https://rubyonrails.org/), one of the most popular web frameworks used across government.
+
+Without these and other projects like it, teams working on GOV.UK services would need to implement all the patterns and behaviour described in the design system themselves. This would be huge a waste of effort and slow down delivery to a crawl.
+
+> Time spent on [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) is time spent not focussing on user needs.
+
+When working on anything that's used by lots of teams, like [referring serious teacher misconduct](https://refer-serious-misconduct.education.gov.uk), [whistleblowing at a company](https://make-a-business-whistleblower-report.service.gov.uk/) or even [GOV.UK forms itself](https://www.forms.service.gov.uk/), there's a great deal of pressure not to break anything. This responsibility forces us to make changes with a degree of caution and to rigorously test and review.
+
+There are two main reasons for us making changes:
+
+* **following upstream changes** --- like adding a new GOV.UK component or adding support for a new version of Ruby or Rails
+* **improvements we feel make the library better** --- like adding features, tidying code or making the tests run faster
```

Comment:
We can use Unicode characters (although I’m constantly in debate with myself about how much Markdown should only use ASCII characters or not)

```suggestion
* **following upstream changes** — like adding a new GOV.UK component or adding support for a new version of Ruby or Rails
* **improvements we feel make the library better** — like adding features, tidying code or making the tests run faster
```